pkg/processor/volume/actualizer.go

```diff
@@ -0,0 +1,79 @@
+package volume
+
+import (
+	"context"
+	"time"
+
+	v1 "github.com/dell/csi-baremetal/api/v1"
+	"github.com/dell/csi-baremetal/api/v1/volumecrd"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const ctxTimeout = 30 * time.Second
+
+// Actualizer is the watcher to update volume statuses
+type Actualizer interface {
+	Handle(ctx context.Context)
+}
+
+type actualizer struct {
+	client client.Client
+	log    *logrus.Entry
+}
+
+func (a *actualizer) Handle(ctx context.Context) {
+	ctx, cancelFn := context.WithTimeout(ctx, ctxTimeout)
+	defer cancelFn()
+
+	volumes := &volumecrd.VolumeList{}
+	if err := a.client.List(ctx, volumes); err != nil {
+		a.log.Errorf("failed to get Volume List: %s", err.Error())
+		return
+	}
+
+	for i := 0; i < len(volumes.Items); i++ {
+		if volumes.Items[i].Spec.CSIStatus == v1.Published && a.ownerPodsAreRemoved(ctx, &volumes.Items[i]) {
```

Comment:
There is no guarantee that volume is unmounted (un-published and un-staged) when pod is removed. When `DeleteVolume` request is received CSI should check for mount status and allow removal if unmounted.

Reply:
When pod is deleted by kubelet and without CSI registered (as it is in our case) - then kubelet unmounts volume by itself:
```
May 30 15:25:52 kind-worker2 kubelet[184]: W0530 15:25:52.792837     184 kubelet_volumes.go:139] Cleaned up orphaned volume from pod "82c5671b-fb1d-4793-9bc0-c76266cda481" at /var/lib/kubelet/pods/82c5671b-fb1d-4793-9bc0-c76266cda481/volumes/kubernetes.io~csi/pvc-58a38b01-03dc-4bcd-a7e8-556e48d6b503/mount
```

Reply:
What if it failed for some reason? You are guessing about volume in state in Operator

Reply:
Operator doesn't manage volumes and doesn't know mount status on nodes, csi node service is in charge of this. By guessing I mean that you rely on internal kubelet logic which is not transparent and might be different on OpenShift, RKE2, etc. I suggest to have mount status check and mount status field in Volume CR. When `DeleteVolume` request is received allow removal if volume is unmounted.

Reply:
And yes - Operator doesn't operate over mount operations and know nothing about it - it operates only Statuses of Volumes in such of corner case which we have.
In either way we need some kind of detection of inconsistency between Volume statuses and actual state at node. It can be performed at 2 ways:
1. As I did with help of operator as initiator of fixing Volume status.
2. Continiously polling at every Node controller such conditions.
The second options is much worse because it is scalable at all with increasing of node count. So, in my opinion the first option, which I did, is preferable.

Reply:
CSI Operator shouldn't manage Volume CRs at all. CSI Node Service must be in charge of Volume mount status detection. CSI Controller must allow deletion of volumes if they are not mounted.